### DISCUSSION TITLE
视频掉帧问题

### DISCUSSION THREAD

**Discussion by ghost** [hooke007/mpv_PlayKit#311] [mpv 通用]
当我播放一个 30fps 的视频或是直播流时， 我不会掉帧， drop frames一直都是0。
  当我播放一个 60fps 的视频或是直播流时，就会出现掉帧情况，drop frames 会一直慢慢得增加。但是任务管理器中，mpv的占用是：cpu 4%左右， gpu 10%左右，可见并没有出现硬件的瓶颈。
  我的显示器是 60hz 1080p 。 cpu是8700，gpu是 nvidia 1660 。 
  没有使用任何的补帧脚本和着色器。
```
我的视频部分配置是：
vo = gpu
hwdec = auto-safe
gpu-api = auto
gpu-context = auto
video-sync = audio
scale = spline36
cscale = spline36
dscale = mitchell
dither-depth = no
correct-downscaling = no
sigmoid-upscaling = no
deband
d3d11va-zero-copy = no
hdr-compute-peak = auto
allow-delayed-peak-detect = yes
```
有没有人知道这是什么情况？

**Comment by hooke007** [comment]
拿本地视频测；
我的建议是从no-config开始加参数或者从现在的配置砍参数观察情况，去判断是哪（/几）个参数引入的问题。
这也是毫无线索的情况下的基操。

当然如果no-config也掉那建议直接给mpv报issue

btw，你的配置里很多都是无效参数（因为等同默认值）写不写都一样

**Reply by ghost** [reply]
谢谢，我用 no-config 和 本地视频 测试了，找到了问题所在。
出现这种情况是因为 用 I 调用内置脚本一直显示统计信息， 统计信息中 drop frames 就会一直增加。
如果 只用 i 短暂显示统计信息，drop frames 就基本不会增加。
如果视频开始时先显示一遍统计信息， 等快播放结束再显示 统计信息， drop frames 不变或是+1 。

这是因为内置脚本显示统计信息 让 mpv 出现丢帧情况吗？

**Reply by hooke007** [reply]
这个问题我没法回答，stats常驻的时候产生的一些读写理论上也不应该产生这么明显的问题，考虑到你的硬件就更不合理了，建议报issue寻找更专业的回答

**Reply by ghost** [reply]
> 这个问题我没法回答，stats常驻的时候产生的一些读写理论上也不应该产生这么明显的问题，考虑到你的硬件就更不合理了，建议报issue寻找更专业的回答

谢谢你的回答，我去 mpv 的issue区问问

**Comment by hooke007** [comment]
友情提醒 https://github.com/mpv-player/mpv/issues/11688
首先你的提问方式不正确，不走模板那几乎就只能当作一般提问。我用过再烂的机器也没有因为常驻stats掉帧，而我刚接触mpv那会用的机器也刚好是8700，我从来没遇到过这种情况。更不要说在我的反馈区里用着不支持v3微架构的也未有报告过此类情况。
其次缺乏细节，片源样本是什么，全面硬件配置是什么都没有提。

因此，你得到的是完全没毛病的官方回答，是的的确影响性能，但是结合我目前看到的条件不合理。
